src/bridge/cockpitdbusjson.c

```diff
@@ -2622,6 +2635,15 @@ cockpit_dbus_json_init (CockpitDBusJson *self)
 
   self->peers = g_hash_table_new (g_str_hash, g_str_equal);
   self->interface_info = cockpit_dbus_interface_info_new ();
+
+  /* create global objects */
+  if (group_connections == NULL)
+    {
+      group_connections = g_hash_table_new_full (g_str_hash, g_str_equal,
+                                                 g_free, g_object_unref);
+      pending_group_connections = g_hash_table_new_full (g_str_hash, g_str_equal,
+                                                         g_free, (GDestroyNotify) g_ptr_array_unref);
```

Comment:
This can't be done in finalize, as other instances might still use the connections. I've added a weak notify on the connection object instead.